lib/spack/spack/cmd/checksum.py

```diff
@@ -56,7 +64,8 @@ def checksum(parser, args):
             tty.die("Could not find any versions for {0}".format(pkg.name))
 
     version_lines = spack.stage.get_checksums_for_versions(
-        url_dict, pkg.name, keep_stage=args.keep_stage)
+        url_dict, pkg.name, keep_stage=args.keep_stage,
+        batch=(args.yes or len(args.versions) > 1))
```

Comment:
```suggestion
        batch=(args.batch or len(args.versions) > 0))
```
Arg was renamed to `batch`. Also, I think we want to auto-batch any time the user supplies any number of versions, even 1.